xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/objects/classes/DBListClass.java

```diff
@@ -594,4 +609,15 @@ public void displayView(StringBuffer buffer, String name, String prefix, BaseCol
             buffer.append(getDisplayValue(prop.getValue(), name, map, context));
         }
     }
+
+    private String getPropertyType(String className, String propertyName, XWikiContext context) throws Exception
+    {
+        PropertyClass pc = null;
+        try {
+            pc = (PropertyClass) context.getWiki().getDocument(className, context).getXClass().get(propertyName);
+        } catch (XWikiException e) {
+            LOGGER.error("Failed to get property [{}] for document [{}]", propertyName, className, e);
+        }
+        return pc.newProperty().getClass().getName();
```

Comment:
This will throw a ``NullPointerException`` if an ``XWikiException`` is caught above. You need to choose between

* falling back on a default property type in case of an ``XWikiException`` or
* let the ``XWikiException`` propagate

In general, there are two main reasons for which you would catch an exception:
* provide a fall back or
* throw a more domain-specific exception that wraps the caught exception.